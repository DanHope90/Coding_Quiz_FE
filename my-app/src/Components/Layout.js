import React from "react";
import PropTypes from "prop-types";
import SideBar from "./SideBar";
import "../Styles/Layout.css";

function Layout({ children }) {
  return (
    <div className="upper-class">
      <div className="sidebar"><SideBar /></div>
      <div>{children}</div>
    </div>
  );
}

Layout.propTypes = {
  children: PropTypes.func.isRequired,
};

export default Layout;
