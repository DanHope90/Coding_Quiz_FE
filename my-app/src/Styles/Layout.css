.upper-class {
    display: flex;
    flex-direction: row;
    height: 100vh;
    align-items: stretch;
    background-color: rgba(65, 162, 41, 0.657);
}