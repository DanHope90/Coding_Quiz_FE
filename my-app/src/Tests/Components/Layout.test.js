import Layout from "../../Components/Layout";
import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";

test("renders the layout component correctly", () => {
    //Wrapping component in BrowserRouter
    const testRenderer = renderer.create(<BrowserRouter><Layout  /></BrowserRouter>);
    expect(testRenderer).toMatchSnapshot();
});
