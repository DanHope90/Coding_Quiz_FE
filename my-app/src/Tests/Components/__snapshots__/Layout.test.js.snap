// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the layout component correctly 1`] = `
<div
  className="upper-class"
>
  <div
    className="sidebar"
  >
    <div
      className="sidebar"
    >
      <ul
        className="sidebar-links"
      >
        <li
          className="sidebar-links-item"
        >
          <a
            className="sidebar-links-link"
            href="/register"
            onClick={[Function]}
          >
            Register 
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 576 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M528 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h480c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm0 400H303.2c.9-4.5.8 3.6.8-22.4 0-31.8-30.1-57.6-67.2-57.6-10.8 0-18.7 8-44.8 8-26.9 0-33.4-8-44.8-8-37.1 0-67.2 25.8-67.2 57.6 0 26-.2 17.9.8 22.4H48V144h480v288zm-168-80h112c4.4 0 8-3.6 8-8v-16c0-4.4-3.6-8-8-8H360c-4.4 0-8 3.6-8 8v16c0 4.4 3.6 8 8 8zm0-64h112c4.4 0 8-3.6 8-8v-16c0-4.4-3.6-8-8-8H360c-4.4 0-8 3.6-8 8v16c0 4.4 3.6 8 8 8zm0-64h112c4.4 0 8-3.6 8-8v-16c0-4.4-3.6-8-8-8H360c-4.4 0-8 3.6-8 8v16c0 4.4 3.6 8 8 8zm-168 96c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64z"
              />
            </svg>
          </a>
        </li>
        <li
          className="sidebar-links-item"
        >
          <a
            className="sidebar-links-link"
            href="/"
            onClick={[Function]}
          >
            Login 
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 640 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M624 448h-80V113.45C544 86.19 522.47 64 496 64H384v64h96v384h144c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16zM312.24 1.01l-192 49.74C105.99 54.44 96 67.7 96 82.92V448H16c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h336V33.18c0-21.58-19.56-37.41-39.76-32.17zM264 288c-13.25 0-24-14.33-24-32s10.75-32 24-32 24 14.33 24 32-10.75 32-24 32z"
              />
            </svg>
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="body"
  />
</div>
`;
